app/src/main/java/com/concordia/mcga/fragments/NavigationFragment.java

```diff
@@ -638,29 +637,41 @@ private void updateSearchUI() {
         if (location != null) {
             AppCompatTextView locationText = (AppCompatTextView)
                     toolbarView.findViewById(R.id.search_location_text);
-            locationText.setText(location.getName());
+            if (location instanceof Building) {
+                locationText.setText(((Building) location).getShortName());
+            } else {
+                locationText.setText(location.getName());
+            }
         }
         if (destination != null) {
             AppCompatTextView destinationText = (AppCompatTextView)
                     toolbarView.findViewById(R.id.search_destination_text);
-            destinationText.setText(destination.getName());
+            if (destination instanceof Building) {
+                destinationText.setText(((Building) destination).getShortName());
+            } else {
+                destinationText.setText(destination.getName());
+            }
         }
 
         if (searchState == SearchState.NONE) {
             locationLayout.setVisibility(View.GONE);
             destinationLayout.setVisibility(View.GONE);
             search.setQueryHint("Enter location...");
+            search.setVisibility(View.VISIBLE);
         } else if (searchState == SearchState.LOCATION) {
             locationLayout.setVisibility(View.VISIBLE);
             destinationLayout.setVisibility(View.GONE);
             search.setQueryHint("Enter destination...");
+            search.setVisibility(View.VISIBLE);
         } else if (searchState == SearchState.DESTINATION) {
             locationLayout.setVisibility(View.GONE);
             destinationLayout.setVisibility(View.VISIBLE);
             search.setQueryHint("Enter location...");
+            search.setVisibility(View.VISIBLE);
         } else { // searchState == SearchState.LOCATION_DESTINATION
             locationLayout.setVisibility(View.VISIBLE);
             destinationLayout.setVisibility(View.VISIBLE);
+            search.setVisibility(View.GONE);
             search.setQueryHint("Search...");
         }
```

Comment:
We hide the search bar in the event that we have both the location and the destination.